client/trino-client/src/main/java/io/trino/client/auth/external/SystemOpenRedirectHandler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.client.auth.external;
+
+import java.io.IOException;
+import java.net.URI;
+
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+
+public class SystemOpenRedirectHandler
+        implements RedirectHandler
+{
+    static final String[] linuxBrowsers = {
+            "xdg-open",
+            "chromium",
+            "google-chrome",
+            "firefox",
+            "mozilla",
+            "opera",
+            "epiphany",
+            "konqueror"};
+
+    @Override
+    public void redirectTo(URI uri)
+            throws RedirectException
+    {
+        String operatingSystem = System.getProperty("os.name").toLowerCase(ENGLISH);
+
+        try {
+            if (operatingSystem.contains("mac")) {
+                Class.forName("com.apple.eio.FileManager").getDeclaredMethod("openURL", new Class<?>[]{String.class})
+                        .invoke(null, new Object[]{uri.toString()});
+            }
+            else if (operatingSystem.contains("windows")) {
+                Runtime.getRuntime().exec("rundll32 url.dll,FileProtocolHandler " + uri.toString());
+            }
+            else {
+                Runtime.getRuntime().exec(new String[]{findLinuxBrowser(), uri.toString()});
+            }
+        }
+        catch (Exception e) {
+            throw new RedirectException(format("Could not open uri %s", uri), e);
+        }
+    }
+
+    private static String findLinuxBrowser()
+    {
+        for (String candidate : linuxBrowsers) {
+            try {
+                if (Runtime.getRuntime().exec(new String[]{"which", candidate}).getInputStream().read() != -1) {
```

Comment:
`which` might not exist.
A more robust option might be to traverse `$PATH` using something like `File.listFiles(FileFilter)`